core/src/main/java/org/infinispan/container/entries/ClusteredRepeatableReadEntry.java

```diff
@@ -82,7 +79,7 @@ private EntryVersion getCurrentEntryVersion(DataContainer container, Persistence
       EntryVersion prevVersion;// on origin, the version seen is acquired without the lock, so we have to retrieve it again
       // TODO: persistence should be more orthogonal to any entry type - this should be handled in interceptor
       InternalCacheEntry ice = PersistenceUtil.loadAndStoreInDataContainer(container, persistenceManager, getKey(),
-            ctx, timeService, ignored);
+            ctx, timeService, null);
```

Comment:
+1 to remove the `AtomicReference` here.

I'd even resurrect the `isLoaded()` flag in `InternalCacheEntry` and use it instead, removing the `AtomicReference` parameter completely. The `isLoaded.get() == Boolean.FALSE` case seems equivalent to `entry == null`, so we don't really need 3 possibilities.